etc/cmake/options.cmake

```diff
@@ -190,8 +192,6 @@ else()
     message(FATAL_ERROR "Invalid max RCP restoration count: ${OT_RCP_RESTORATION_MAX_COUNT}")
 endif()
 
-option(OT_EXCLUDE_TCPLP_LIB "exclude TCPlp library from build")
-
```

Comment:
Thinking more on this, I am not sure if it is good idea to remove `OT_EXCLUDE_TCPLP_LIB` and use `OT_TCP` as the way to decide whether to include the `tcplp` library or not.

In general, we provide different ways for project integrators to configure OpenThread:
1. They can use CMake options to enable/disable features during build.
2. Or we allow then to  define the related `OPENTHREAD_COFNIG_{}` in a project/product-specific config header file (this is the model I personally recommend for products).

All CMake options have 3 states: `ON`, `OFF`, or (empty) not specified. The "not specified" state allows the second way above (using project-specific config header).


With the change in this PR, we are indirectly forcing the users to enable/disable TCP feature using `CMake` option as the only way.

Reply:
I think would be good for @edmont  to review.

Removing this `if(NOT OT_EXCLUDE_TCPLP_LIB)` check always include the lib. I think  there was some situation where this was not desirable and reason for `OT_EXCLUDE_TCPLP_LIB` to be added from #7306.

Reply:
Hi, we have discussed this internally and this is not an issue anymore, so I'm fine with removing `OT_EXCLUDE_TCPLP_LIB`.